### DISCUSSION TITLE
ITK image filter as a MONAI transform?

### DISCUSSION THREAD

**Discussion by dzenanz** [Project-MONAI/MONAI#6098] [Ideas]
@thewtex had an idea of creating MONAI transform which would call ITK's image filters. The transform would take the ITK filter name, and a dictionary of filter's parameters as its own parameters. I would be mostly implementing it.

What do you think of the idea? Any remarks on its feasibility, or effort level?

**Comment by dzenanz** [comment]
Any opinions or advice?

**Reply by wyli** [reply]
there are examples of integrating ITK image filters https://github.com/Project-MONAI/tutorials/blob/main/modules/integrate_3rd_party_transforms.ipynb 

any specific use cases you are considering?

**Reply by dzenanz** [reply]
The tutorial is quite nice and helpful. Using ITK filters is possible and relatively easy. However, we could make it easier by building on the tutorial: handling ITK notes below and avoiding the need to write a dedicated function for each filter.

ITK Notes:
1. ITK transform function API has several args(not only data), need to set args in wrapper before Compose.
2. If input data is Numpy, ITK can't support dict type, need wrapper to convert the format.
3. ITK expects input shape [H, W, [D]], so handle every channel and stack the results.

**Reply by wyli** [reply]
I see, please feel free to start a pull request, I can help review and merge the feature.
